app/src/app/store/packets/thunks.ts

```diff
@@ -13,6 +20,13 @@ export enum PacketsMutationType {
 }
 
 export const actions: PacketsActions = {
-    fetchPackets: api.get<Packet[], void>(PacketsMutationType.GetPackets, "/packets"),
-    fetchPacketById: api.get<PacketMetadata, string>(PacketsMutationType.GetPacket, "/packets/metadata")
+    fetchPackets: createAsyncThunk<PageablePackets, PaginationProps, CustomAsyncThunkOptions>(
+        PacketsMutationType.GetPackets,
+        async (queryParams, thunkAPI) =>
+            api.get(`/packets/?${qs.stringify(queryParams)}`, thunkAPI)),
```

Comment:
Well, this is how it should be. I considered several options and scenarios before deciding to refactor it. For example, if we have an endpoint such as `/metadata/{packetId}/changelog?day=monday` then the initial implementation becomes complex and unreadable, especially when sending some this like `{packetId}/changelog?day=monday` as a parameter.

Reply:
Could we talk about it tomorrow when you're in? I would like to know more about the refactor options and why the previous pattern no longer works (mostly because I'm quite unfamiliar with modern redux). I wasn't suggesting sending `{packetId}/changelog?day=monday` as a parameter! Sure, we could have a number of parameters determining the data to be fetched from the API, but I think those would ideally be abstracted into an action payload for use by the component since otherwise the knowledge of the API is spread between the components and the actions.

Reply:
> As it stands now, we can pass value down for queryParam and path variable. for each unique endpoint without the need to start filtering wether a request should have pathVariable or/and queryParams. 

You could do both or either, or more (e.g. multiple path variables or values to be passed in post bodies), with payload types as we already do HINT. I don't think there's anywhere in HINT where the component explicitly knows the API details and considers payload values to be qs parameters (although, actually, here, it's still the action appending that qs to the url, so it could potentially do something else with those e.g. if the API changed and those were now embedded in the path), so I think I partly just have an issue with parameter naming in the action.. 

I'll have a go at putting together the approach I mean in a branch - it may be identical to one you've already rejected, but then we'll be sure we're not talking at cross purposes!